opensrp-chw-core/src/main/java/org/smartregister/chw/core/dao/ChildDao.java

```diff
@@ -111,6 +116,84 @@ public static boolean isChild(String baseEntityID) {
         return res.get(0) > 0;
     }
 
+    public static String getBaseEntityID(String identifierType, String id) {
+        String[] projectionArgs = new String[]{BASE_ENTITY_ID};
+        SQLiteDatabase database = getRepository().getReadableDatabase();
+        net.sqlcipher.Cursor cursor = null;
+        try {
+            if (database == null) {
+                return null;
+            }
+            String selection = "select base_entity_id from ec_child where ? = ?";
+            String[] selectionArgs = new String[]{identifierType, id};
+
+
+            cursor = database.query(CoreConstants.TABLE_NAME.CHILD, projectionArgs, selection, selectionArgs, null, null, null);
+
+            if (cursor != null && cursor.getCount() > 0 && cursor.moveToFirst()) {
+                return cursor.getString(cursor.getColumnIndex(BASE_ENTITY_ID));
+            }
+        } catch (Exception e) {
+            Timber.e(e);
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+        return null;
+    }
+
+    public static String queryColumnWithEntityId(String baseEntityId, String columnName) {
```

Comment:
make the function getBaseEntityID accept query, params, column  name and resuse so that there is no duplicated code

Reply:
Can you please elaborate on this part? We have different methods for different types of queries.

Reply:
I suggest you use the same function for getBaseEntityID,isThinkMDCarePlanExist and queryColumnWithEntityId by creating a function that allows you pass query and params and return a value for column name

Reply:
Ideally you can use `queryColumnWithEntityId` in `getBaseEntityID` and `isThinkMDCarePlanExist` functions